Added new DoD topics #9282 (#9)
AdobeDocs/commerce-contributor: src/pages/guides/code-contributions/dod-backward-compat.md

---
title: Definition of Done - Backwards Compatibility | Commerce Contributor

> description: Backwards compatibility in Commerce projects.

Reviewer: Please rewrite following guidelines.

---

> ## Overview

Reviewer: ```suggestion
# Backward compatibility definition of done
```

All backward incompatible changes should be approved and covered by static tests to meet the "Definition of Done" (DoD).

> ## Backward Compatibility

Reviewer: ```suggestion
```

> ### Functional Backward Compatibility

Reviewer: ```suggestion
## Functional backward compatibility
```

**Functional backward compatibility** means the behavior of the application is preserved.

-  Existing product features and functionality must be retained during any changes to the code.
-  Any backward-incompatible functional changes must be approved by a product owner.
-  The documentation should explain the justification and provide a business value.

> ### Technical Backward Compatibility

Reviewer: ```suggestion
## Technical backward compatibility
```

**Technical backward compatibility** means the technical interfaces are preserved.

Technical interfaces include PHP interfaces or classes, CLI commands, URLs, or any other interfaces that can be used by 3rd-party developer, system integrator, or user of Magento.
Any change to an interface that can lead to a broken integration is a breaking technical change.

> Technical interfaces and the corresponding level of change are described in [Code Changes][0] and [Module Version Dependencies][1].

Reviewer: ```suggestion
Technical interfaces and the corresponding level of change are described in [Code Changes](https://devdocs.magento.com/guides/v2.4/extension-dev-guide/versioning/codebase-changes.html) and [Module Version Dependencies](https://devdocs.magento.com/guides/v2.4/extension-dev-guide/versioning/dependencies.html).
```

Technical backward compatibility must be retained between PATCH (marketing) versions of Magento products. It should also be retained between MINOR (marketing) releases if possible.
Any breaking changes must be approved by an architect, product owner and release manager.

> For more information, see [Magento's backward compatibility policy][2] and [Versioning][3] documents.

Reviewer: ```suggestion
For more information, see [Backward compatibility policy](backward-compatibility-policy.md) and [Versioning](https://devdocs.magento.com/guides/v2.4/extension-dev-guide/versioning/).
```

> [0]:https://devdocs.magento.com/guides/v2.4/extension-dev-guide/versioning/codebase-changes.html

Reviewer: ```suggestion
```

> [1]:https://devdocs.magento.com/guides/v2.4/extension-dev-guide/versioning/dependencies.html

Reviewer: ```suggestion
```

> [2]:./backward-compatibility-policy.md

Reviewer: ```suggestion
```

> [3]:https://devdocs.magento.com/guides/v2.4/extension-dev-guide/versioning/

Reviewer: ```suggestion
```
